src/gui/TotpExportSettingsDialog.cpp

```diff
@@ -29,25 +29,28 @@
 #include <QMessageBox>
 #include <QPushButton>
 #include <QShortcut>
+#include <QStackedWidget>
 
 TotpExportSettingsDialog::TotpExportSettingsDialog(DatabaseWidget* parent, Entry* entry)
     : QDialog(parent)
     , m_timer(new QTimer(this))
     , m_verticalLayout(new QVBoxLayout())
-    , m_totpSvgWidget(new SquareSvgWidget())
+    , m_totpSvgContainerWidget(new QStackedWidget())
```

Comment:
Why are you using a stacked widget?

Reply:
I was looking for Frame based widget that can include other widget(s). Only stacked widget I found anong standard QT ones.

Reply:
You don't need to wrap it in a frame, it's part of a layout. You really just need to get the sizeHint() of the svg widget after a resize and then force it to be square. The size hint returns the size the layout wants it to be.